### PR TITLE
crd/ContourDeployment: Add field 'podAnnotations' for contour

### PR DESCRIPTION
**Pull request by izturn**
add the field podAnnotations to crd/ContourDeployment to enable adding more annotations to pod/contour

Use case: istio injection en/disabled 

Signed-off-by: Gang Liu [gang.liu@daocloud.io](mailto:gang.liu@daocloud.io)

### CODE REVIEW

**Comment by izturn** [comment]
ping @skriss @sunjayBhatia

File: internal/provisioner/objects/deployment/deployment.go [hunk 1]
```diff
@@ -301,3 +298,32 @@ func contourPodLabels(contour *model.Contour) map[string]string {
...
+}
+
+/*
+	Annotations: map[string]string{
+		"prometheus.io/scrape": "true",
+		"prometheus.io/port":   fmt.Sprintf("%d", metricsPort),
+	},
+*/
+
+//					Annotations: envoyPodAnnotations(contour),
```
Comment (skriss) [inline internal/provisioner/objects/deployment/deployment.go:329]: please remove these lines

**Review by skriss — CHANGES_REQUESTED** [review]
one nit, otherwise LGTM

File: internal/provisioner/controller/gateway_test.go [hunk 2]
```diff
@@ -1267,6 +1267,52 @@ func TestGatewayReconcile(t *testing.T) {
 			},
 		},
 
+		"If ContourDeployment.Spec.Contour.PodAnnotations is specified, the Contour pods' have annotations for prometheus & user-defined": {
```
Comment (sunjayBhatia) [inline internal/provisioner/controller/gateway_test.go:1270]: doesn't look like this test checks for the prometheus annotations as stated here?

File: internal/provisioner/objects/deployment/deployment.go [hunk 3]
```diff
@@ -301,3 +298,32 @@ func contourPodLabels(contour *model.Contour) map[string]string {
...
+	}
+
+	metricsPort := metricsPort
+	if contour.Spec.RuntimeSettings != nil &&
+		contour.Spec.RuntimeSettings.Metrics != nil &&
+		contour.Spec.RuntimeSettings.Metrics.Port > 0 {
+		metricsPort = contour.Spec.RuntimeSettings.Metrics.Port
+	}
+
+	annotations["prometheus.io/scrape"] = "true"
```
Comment (sunjayBhatia) [inline internal/provisioner/objects/deployment/deployment.go:316]: worth testing+documenting that if the user tries to set these, then we will overwrite them with our values

File: internal/provisioner/controller/gateway_test.go [hunk 4]
```diff
@@ -1263,7 +1263,54 @@ func TestGatewayReconcile(t *testing.T) {
...
+				},
+				Spec: contourv1alpha1.ContourDeploymentSpec{
+					Contour: &contourv1alpha1.ContourSettings{
+						PodAnnotations: map[string]string{
+							"key": "val",
+						},
+					},
+				},
+			},
+			gateway: &gatewayv1beta1.Gateway{
```
Comment (sunjayBhatia) [inline internal/provisioner/controller/gateway_test.go:1286]: i think your other refactor PR `makeGateway` helper could be useful here

Comment (izturn) [reply internal/provisioner/controller/gateway_test.go:1286]: yes,  this pr leads to that one, i will merge & refactor this to sync with it

**Review by sunjayBhatia — APPROVED** [review]

**Review by skriss — APPROVED** [review]

**Pull request merged**